sechub-scan/src/test/java/com/mercedesbenz/sechub/domain/scan/ScanTypeCountTest.java

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+package com.mercedesbenz.sechub.domain.scan;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.mercedesbenz.sechub.commons.model.ScanType;
+
+public class ScanTypeCountTest {
+
+    private ScanTypeCount scanTypeCount;
+
+    @BeforeEach
+    void beforeEach() {
+        scanTypeCount = ScanTypeCount.of(ScanType.CODE_SCAN);
+    }
+
+    @Test
+    void of_constructor_creates_right_object() {
+        /* prepare + execute */
+        scanTypeCount = ScanTypeCount.of(ScanType.CODE_SCAN);
+
+        /* test */
+        assertEquals(scanTypeCount.getScanType(), ScanType.CODE_SCAN);
+        assertEquals(scanTypeCount.getHighSeverityCount(), 0);
+        assertEquals(scanTypeCount.getMediumSeverityCount(), 0);
+        assertEquals(scanTypeCount.getLowSeverityCount(), 0);
+    }
+
+    @Test
+    void when_ScanType_is_null_then_of_constructor_throws_IllegalArgumentException() {
+        /* prepare + execute */
+        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
+            ScanTypeCount.of(null);
+        });
+        String expectedMessage = "ScanType argument must exist";
```

Comment:
this should be in the prepare section of the tests.

Reply:
Done.